Apps/Playground/Scripts/experience.js

```diff
@@ -83,6 +84,30 @@ CreateBoxAsync(scene).then(function () {
         }, { maxWidth: 256, maxHeight: 256, facingMode: "environment" });
     }
 
+    if (readPixels) {
+        const texture = new BABYLON.Texture("https://assets.babylonjs.com/textures/earth.jpg", scene);
+        texture.onLoadObservable.addOnce(() => {
+            const mip = 1;
+            const textureWidth = texture.getSize().width >> mip;
+            const textureHeight = texture.getSize().height >> mip;
+            const x = textureWidth / 4;
+            const y = textureHeight / 4;
+            const width = textureWidth / 2;
+            const height = textureHeight / 2;
+            // This read will create a new buffer.
+            texture.readPixels(undefined, mip, undefined, undefined, undefined, x, y, width, height).then((buffer) => {
+                console.log(`Read ${buffer.byteLength} pixel bytes.`);
+                return buffer;
+            })
+            .then(buffer => {
+                // This read reuses the existing buffer.
+                texture.readPixels(undefined, mip, buffer, undefined, undefined, x, y, width, height).then((buffer) => {
+                    console.log(`Read ${buffer.byteLength} pixel bytes.`);
+                });
+            });
+        });
+    }
```

Comment:
I don't like this pattern, but it's okay for now. We need to move these into jest tests once jest is ready to use.